web/service_agreements_controller.go

```diff
@@ -30,7 +31,12 @@ func (sac *ServiceAgreementsController) Create(c *gin.Context) {
 	} else if err = sac.App.Store.SaveServiceAgreement(&sa); err != nil {
 		c.AbortWithError(500, err)
 	} else {
-		c.JSON(200, sa)
+		buffer, err := NewJSONAPIResponse(&sa)
+		if err != nil {
```

Comment:
I'm not sure what value `NewJSONAPIResonse` adds over this:

```golang
func (jrc *JobRunsController) Show(c *gin.Context) {
	id := c.Param("RunID")
	if jr, err := jrc.App.Store.FindJobRun(id); err == storm.ErrNotFound {
		c.AbortWithError(404, errors.New("Job Run not found"))
	} else if err != nil {
		c.AbortWithError(500, err)
	} else if doc, err := jsonapi.Marshal(presenters.JobRun{jr}); err != nil {
                 // same as NewJSONAPIResponse?
		c.AbortWithError(500, err)
	} else {
		c.Data(200, MediaType, doc)
	}
}
```

Is it for the more explicit error message?

If possible, I think it's clearer to have `else if ...` rather than a nested `if`.

Reply:
I personally prefer the `NewJSONAPIResponse` it's more semantic and reduces the complexity of the calling function. Will reduce the if nesting and revisit.